### PR TITLE
He Initialization and Lecun Normal initialization

### PR DESCRIPTION
**Pull request by Prabhat-IIT**
It is in continuation with #1280 
I've just corrected minor style mistakes. I think these initializations are need of the hour and should be incorporated as soon as possible.

### CODE REVIEW

**Comment by Prabhat-IIT** [comment]
@zoq can you run the travis again?

**Comment by Prabhat-IIT** [comment]
@zoq style check is failing due to `svd_impl`. Same is the case with SAGA pr. I think it'll need a separate commit.

**Comment by zoq** [comment]
@mlpack-jenkins test this please

File: src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
+/**
+ * @file lecun_normal_init.hpp
+ * @author Dakshit Agrawal and Prabhat Sharma
```
Comment (zoq) [inline src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp]: Can you split the authors (use @author for each name).

File: src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+   * @param rows Number of rows.
+   * @param cols Number of columns.
+   */
+  void Initialize(arma::mat& W,
+                  const size_t rows,
+                  const size_t cols)
+  {
+    // He initialization rule says to initialize weights with random
+    // values taken from a gaussian distribution with mean = 0 and
+    // standard deviation = sqrt(1/rows), i.e. variance = (1/rows).
```
Comment (zoq) [inline src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp]: Super picky style comment, can you use a space between operations?

File: src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    const double variance = 1.0 / ((double) rows);
+
+    if (W.is_empty())
+    {
+      W.set_size(rows, cols);
+    }
+
+    // Multipling a random variable X with variance V(X) by some factor c,
+    // then the variance V(cX) = (c^2)* V(X).
```
Comment (zoq) [inline src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp]: See comment above.

File: src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * This class is used to initialize weight matrix with the Lecun Normalization
+ * initialization rule.
+ *
+ * For more information, the following papers can be referred to:
+ *
+ * @code
+ * @inproceedings{conf/nips/KlambauerUMH17,
```
Comment (zoq) [inline src/mlpack/methods/ann/init_rules/lecun_normal_init.hpp]: Do you mind to reformat the citation, here is an example: 
https://github.com/mlpack/mlpack/blob/b6b96de5df581f9cf7d4d0db1015d7fbb685fee4/src/mlpack/core/optimizers/adam/adam_update.hpp#L32-L40

File: src/mlpack/methods/ann/init_rules/he_init.hpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
+/**
+ * @file he_init.hpp
+ * @authors Dakshit Agrawal and Prabhat Sharma
```
Comment (zoq) [inline src/mlpack/methods/ann/init_rules/he_init.hpp]: Can you split up the author section (for each author another line that starts with ``@author``), that way doxygen can pick this up.

File: src/mlpack/methods/ann/init_rules/he_init.hpp [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+/**
+ * This class is used to initialize weight matrix with the He
+ * initialization rule given by He et. al. for neural networks. The He
+ * initialization initializes weights of the neural network to better
+ * suit the rectified activation units.
+ *
+ * For more information, the following paper can be referred to:
+ *
+ * @code
+ * @article{He2015DelvingDI,
```
Comment (zoq) [inline src/mlpack/methods/ann/init_rules/he_init.hpp]: Do you mind to reformat this as well?

**Comment by dakshitagrawal97** [comment]
Thanks @Prabhat-IIT for completing this PR!  Appreciate it.  :)

**Comment by Prabhat-IIT** [comment]
@rcurtin the citations have been already formatted :)

**Comment by rcurtin** [comment]
Can you match the indentation of the citations @zoq provided as an example please?

**Comment by Prabhat-IIT** [comment]
Sorry, @rcurtin  I didn't see the minute spacing details :(

**Comment by Prabhat-IIT** [comment]
@zoq travis failed once due to time out. I think you can merge this after initiating travis build once again

**Comment by zoq** [comment]
Thanks again for another great contribution.

**Pull request merged**